[lldb][sbdebugger] Move SBDebugger Broadcast bit enum into lldb-enumerations.h

chelcassanova opened the pull request:
When the `eBroadcastBitProgressCategory` bit was originally added to Debugger.h and SBDebugger.h, each corresponding bit was added in order of the other bits that were previously there. Since `Debugger.h` has an enum bit that `SBDebugger.h` does not, this meant that their offsets did not match. 

Instead of trying to keep the bit offsets in sync between the two, it's preferable to just move SBDebugger's enum into the main enumerations header and use the bits from there.

chelcassanova committed:
[lldb][sbdebugger] Match progress category enum bit in Debugger.h

When the `eBroadcastBitProgressCategory` bit was originally added to
Debugger.h and SBDebugger.h, each corresponding bit was added in order
of the other bits that were previously there. Since `Debugger.h` has an
enum bit that `SBDebugger.h` does not, this meant that their offsets did
not match. This commit changes it so that the bit offsets match each other.

JDevlieghere commented on lldb/include/lldb/API/SBDebugger.h:
```diff
@@ -46,7 +46,8 @@ class LLDB_API SBDebugger {
       eBroadcastBitError = (1 << 2),
-      eBroadcastBitProgressCategory = (1 << 3),
+      eBroadcastBitProgressCategory =
+          (1 << 4), // This needs to match the corresponding bit in Debugger.h
```
Let's move that comment above the enum.

JDevlieghere commented on lldb/include/lldb/API/SBDebugger.h:
```diff
@@ -46,7 +46,8 @@ class LLDB_API SBDebugger {
       eBroadcastBitWarning = (1 << 1),
       eBroadcastBitError = (1 << 2),
-      eBroadcastBitProgressCategory = (1 << 3),
+      // This needs to match the corresponding bit in Debugger.h
```
Doesn't this comment apply to the whole enum? My suggestion was to move this above line 45 to convey that, unless that's not the case?

chelcassanova replied:
My mistake, I was indicating that the progress category bit is skipping `(1 << 3)` in this sequence since that bit has to match the one in `Debugger.h`. I can change the comment to say that the whole enum needs to match `Debugger.h`.

medismailben commented:
Why don't we just remove the enum from SBDebugger and use the one from Debugger instead, this will ensure the offsets are always the same.

bulbazord commented:
> Why don't we just remove the enum from SBDebugger and use the one from Debugger instead, this will ensure the offsets are always the same.

SBDebugger is public and Debugger is private, so I'm not sure how you can do this? Maybe I'm misunderstanding your suggestion...

It would be nice if we could move the enum into `lldb-enumerations` so we didn't have to worry about keeping this in sync though.

chelcassanova committed:
Add SBDebugger broadcast bits to lldb-enumerations

Per Alex's suggestions, adds the broadcast bits from SBDebugger
into lldb-enumerations.

JDevlieghere commented on lldb/include/lldb/API/SBDebugger.h:
```diff
@@ -42,11 +42,13 @@ class LLDB_API SBInputReader {
 class LLDB_API SBDebugger {
 public:
+  // The enum values here need to match their corresponding values in
+  // Debugger.h.
   FLAGS_ANONYMOUS_ENUM(){
       eBroadcastBitProgress = (1 << 0),
       eBroadcastBitWarning = (1 << 1),
       eBroadcastBitError = (1 << 2),
-      eBroadcastBitProgressCategory = (1 << 3),
+      eBroadcastBitProgressCategory = (1 << 4),
```
This one can go now, right?

chelcassanova replied:
Yup, will remove.

JDevlieghere commented on lldb/include/lldb/lldb-enumerations.h:
```diff
@@ -1339,6 +1339,14 @@ enum AddressMaskRange {
 };
 
+/// Used by the debugger to indicate which events are being broadcasted.
+enum DebuggerBroadcast {
```
`DebuggerBroadcastBit` maybe?

chelcassanova committed:
Remove broadcast enum from SBDebugger

Removes the enum for broadcast bits from SBDebugger as they're now in
lldb-enumerations.h. This also requires uses of these bits to be updated
in some API tests.

medismailben reviewed (approved):
LGTM if you address the comment and rewrite the PR description and title :)

chelcassanova commented on lldb/include/lldb/lldb-enumerations.h:
```diff
@@ -1339,6 +1339,14 @@ enum AddressMaskRange {
+  eBroadcastBitProgress = (1 << 0),
+  eBroadcastBitWarning = (1 << 1),
+  eBroadcastBitError = (1 << 2),
+  eBroadcastBitProgressCategory = (1 << 4),
```
Oh yeah, I guess since this is in `lldb-enumerations` it doesn't have to match the ones in Debugger.h anymore.
